docker_cli/_image.py

```diff
@@ -0,0 +1,471 @@
+            f"--network={network}",
+            context_str,
+            "-t",
+            tag
+        ]
+
+        if no_cache:
+            cmd += ["--no-cache"]
+
+        if dockerfile_build:
+            # If a dockerfile path is given, include it.
+            # Else, docker build will default to "./Dockerfile"
+            if dockerfile is not None:
+                cmd += [f"--file={os.fspath(dockerfile)}"]
+            stdin = None
+        else:
+            cmd += ["-f-"]
+            stdin = dockerfile_string
+
+        try:
+            run(
+                cmd,
+                text=True,
+                stdout=stdout,  # type: ignore
+                stderr=stderr,  # type: ignore
+                input=stdin,
+                check=True
+            )
+        except CalledProcessError as err:
+            if dockerfile_build:
+                raise DockerBuildError(tag, dockerfile) from err
+            else:
+                raise DockerBuildError(
+                    tag,
+                    dockerfile_string,
+                    f"String dockerfile {tag} failed to build."
+                ) from err
+
+        return cls(tag)
+
+    def _inspect(self, format: Optional[str] = None) -> str:
+        """
+        Use 'docker inspect' to retrieve a piece of information about the
+        image.
+
+        Parameters
+        ----------
+        format : str, optional
+            The value to be requested by the --format argument, or None.
+            Defaults to None.
+
+        Returns
+        -------
+        str
+            The string returned by the docker inspect command.
+        """
+        cmd = ["docker", "inspect"]
+        if format:
+            cmd += [f"-f={format}"]
+        cmd += [self._id]
+
+        inspect_result = run(
+            cmd,
+            capture_output=True,
+            text=True,
+            check=True
+        )
+
+        output_text = inspect_result.stdout
+        return output_text
+
+    def run(
+        self,
+        command: str,
+        *,
+        stdout: Optional[Union[io.TextIOBase, int]] = None,
+        stderr: Optional[Union[io.TextIOBase, int]] = None,
+        interactive: bool = False,
+        network: str = "host"
+    ):
+        """
+        Run the given command on a container.
+
+        .. warning::
+            This method does not work correctly if the image built does not have
+            bash installed.
+
+        Parameters
+        ----------
+        cmd : str
+            The desired command, in linux command format.
+        stdout : io.TextIOBase or special value, optional
+            For a description of valid values, see :func:`subprocess.run`.
+        stderr : io.TextIOBase or special value, optional
+            For a description of valid values, see :func:`subprocess.run`.
+        interactive : bool, optional
+            A boolean describing whether or not to run this command in
+            interactive mode or not. Defaults to True.
```

Comment:
Same here.

What probably happened was that you accepted these suggestions via the GitHub UI but then forgot to sync up your local clone with the remote changes. Then at some point later you probably merged the local & remote branches and accidentally reverted the remote changes.

When people make suggestions on my PRs, I usually just make the corresponding edits to my local clone rather than "accepting" the suggestion via the GitHub UI, because it just seems easier to keep all of my changes in one place, but you're of course welcome to do whatever works best for you.

```suggestion
            interactive mode or not. Defaults to False.
```